pkg/providers/gcp/provider_redis_test.go

```diff
@@ -148,12 +148,24 @@ func TestRedisProvider_deleteRedisCluster(t *testing.T) {
 		wantErr bool
 	}{
 		{
-			name: "failed to retrieve gcp redis strategy config",
+			name: "success triggering redis deletion for an existing instance",
 			fields: fields{
 				Client: func() client.Client {
 					mc := moqClient.NewSigsClientMoqWithScheme(scheme)
-					mc.GetFunc = func(ctx context.Context, key k8sTypes.NamespacedName, obj runtime.Object) error {
-						return fmt.Errorf("generic error")
+					mc.GetFunc = func(ctx context.Context, key k8sTypes.NamespacedName, obj client.Object) error {
+						switch cr := obj.(type) {
+						case *v1.Infrastructure:
+							cr.Status.PlatformStatus = &v1.PlatformStatus{
+								GCP: &v1.GCPPlatformStatus{
+									ProjectID: "projectID",
+									Region:    "region",
+								},
+								Type: v1.GCPPlatformType,
+							}
+						case *corev1.ConfigMap:
+							cr.Data = map[string]string{"redis": `{"development":{}}`}
+						}
+						return nil
```

Comment:
After our discussion on unit testing did we not agree on using the moq client with a set of objects instead of explicitly overriding the functions? I thought the intention was to override explicitly only when necessary (i.e. k8s api errors)

Reply:
One minor note on the resolution here is perhaps we should define the duplicated code in functions instead of re-defining in each individual test. For example you can use [this function](https://github.com/adam-cattermole/cloud-resource-operator/blob/MGDAPI-4734/pkg/providers/gcp/cluster_network_provider_test.go#L63-L80) for a valid infrastructure CR. This eases maintainability, as if something changes with the CR we only need to change it in one location instead of for each test. It also makes writing new tests less error prone.

Reply:
Sorry for the nitpicking - thanks for the changes